python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -71,20 +71,17 @@
 from .partition_key_range import PartitionKeyRange
 
 
-def is_second_ambiguous_time(dt: datetime, tz: Optional[str]):
+def is_second_ambiguous_time(dt: datetime, tz: str):
     """Returns if a datetime is the second instance of an ambiguous time in the given timezone due
     to DST transitions.
     """
     # UTC is never ambiguous
-    if tz is not None and tz.upper() == "UTC":
+    if tz.upper() == "UTC":
```

Comment:
Where are some of these changes coming from? I know the eng team did some pendulum stuff recently; was this a bad pull from `master`?

Reply:
Oh good question, I have no idea what this nonsense is

Reply:
The only thing I did was run ruff. I'm not sure what these code changes are.